### ISSUE TITLE
error property not found on scene.py:161

### ISSUE THREAD

**Issue by ivanpicas** [Naxela/The_Lightmapper#40]
Hello, I recently installed this addon that is very useful to me and I have detected this error, which does not allow me to use the denoise neither from nvidia or from optix.

Sorry for my poor english.

![image](https://user-images.githubusercontent.com/68562852/91336705-9e1bec80-e7d2-11ea-9cca-d08befe1b019.png)

**Comment by Naxela** [comment]
Hi and thanks for trying out the addon

I realized I hadn't yet implemented the interface for using Nvidia Optix yet into the 0.3 branch, but now it should be working in the latest commit: https://github.com/Naxela/The_Lightmapper/commit/b62155d26cb91be8cc88c5e1ac9b76205e8df479

Get the latest git version here: https://github.com/Naxela/The_Lightmapper/archive/master.zip

I only tested on Windows using the 2.4 denoiser version from DeclanRussell here: https://github.com/DeclanRussell/NvidiaAIDenoiser/releases/tag/2.4

I hope it works now

**Comment by ivanpicas** [comment]
Thank you very much for the speed. I am trying to bake the lightmaps of a scene with many objects for webgl and your addon is great for me. I will answer you shortly if everything goes well or if something fails.

Greetings and thanks

**Comment by ivanpicas** [comment]
Hello Alexander. I keep trying your addon but now I get another series of errors and I really don't know where they can come from. The objects have nothing but a material and a UV mapping. I am attaching the .blend because I really don't know what happens, the other day I tried a smaller scene with objects and materials created in substance and more or less everything went well but now I have been blocked.

This is the error:

![image](https://user-images.githubusercontent.com/68562852/91408954-00add080-e845-11ea-87f4-8c0ae4543f26.png)


[salon.zip](https://github.com/Naxela/The_Lightmapper/files/5134651/salon.zip)


Regards

**Comment by Naxela** [comment]
Hi, I've had that problem before without being sure what causes it. I'll try to see if I can fix it on a few hours. Thank you for the blend file

Regards

**Comment by ivanpicas** [comment]
You're welcome, thanks to you.
I'm baking the scene little by little until I detect where the error comes from, hopefully luck.

Regards

**Comment by ivanpicas** [comment]
Hello Alexander.

I have been working with your addon and I can already tell you some things that cause the errors.
In the first place, it gave an error because I had the denoiser path configured wrong, so that's my fault.

![image](https://user-images.githubusercontent.com/68562852/91438182-3bc2fa80-e86b-11ea-8b39-1738aac70d36.png)


Second, when it detects an instance the addon gives an error and stops, this can be easily corrected by breaking the links, no problem.
Another error that causes the addon to stop is when it detects an object with glass material, that's my thing, obviously, the glass does not have to be baked.
Another error that I have detected is the second screenshot that I put, which does not generate the connections in the nodes at the end of the bake.

![image](https://user-images.githubusercontent.com/68562852/91438214-467d8f80-e86b-11ea-8195-1dd13e588a55.png)


In summary, the errors that I have detected are mainly due to the configuration or the objects or materials of my scene. What I have done in the end is to apply a base material, a simple principled and eliminate all the instances, from there, the addon has worked perfectly, except the connections of the nodes.

So, congratulations because I think you have done an excellent job and made a very useful addon.

Regards

**Comment by Naxela** [comment]
Hi Ivan,

Thanks for your blend file, I had a look through it and ran into the same errors that you mentioned, although most were setup based it's still something I would like the addon to safeguard against. The NoneType error was caused by an object without a material, the addon now provides a default material now. - Then I ran into a bug in Blender itself, where it wouldn't bake or unwrap if the objects were hidden (such as CORTINA01.low.7). Your file have been very helpful in finding and safeguarding against these situations.

In any case, now your blend should work without errors with the latest commit: https://github.com/Naxela/The_Lightmapper/commit/3b3214fd65d93a6e40c17c6f3b2baaf2bdadeb07

As for glass I didn't know that was also a problem - It it made with the "Glass BSDF" node, or a principled node with alpha? - I'm not sure if Glass is supported as a node, but both metallic and glass materials are something I plan to add some automatic things too.

Hope it helps!

Regards

**Comment by ivanpicas** [comment]
Hi Alexander,

Sure it helps, thank you very much.
The glass is made with principled and a transparency node, something very simple, in principle it does not matter too much because the glass is made a posteriori in unreal or playcanvas or whatever.

I thank you for the quick responses, I will follow your progress closely, you have made a very very useful addon.

Regards!

**Comment by Naxela** [comment]
Hi,

Thank you for your help in regards to identifying these bugs. - If you stumble upon any other bugs, feel free to create a new ticket. Glad it helped

Regards
